smack-experimental/src/main/java/org/jivesoftware/smackx/blocking/BlockingCommandManager.java

```diff
@@ -0,0 +1,177 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.blocking;
+
+import java.util.List;
+import java.util.Map;
+import java.util.WeakHashMap;
+
+import org.jivesoftware.smack.ConnectionCreationListener;
+import org.jivesoftware.smack.Manager;
+import org.jivesoftware.smack.SmackException.NoResponseException;
+import org.jivesoftware.smack.SmackException.NotConnectedException;
+import org.jivesoftware.smack.XMPPConnection;
+import org.jivesoftware.smack.XMPPConnectionRegistry;
+import org.jivesoftware.smack.XMPPException.XMPPErrorException;
+import org.jivesoftware.smack.filter.IQReplyFilter;
+import org.jivesoftware.smack.filter.StanzaFilter;
+import org.jivesoftware.smack.packet.IQ;
+import org.jivesoftware.smack.packet.IQ.Type;
+import org.jivesoftware.smackx.blocking.element.BlockContactIQ;
+import org.jivesoftware.smackx.blocking.element.BlockListIQ;
+import org.jivesoftware.smackx.blocking.element.BlockingCommandElements;
+import org.jivesoftware.smackx.blocking.element.GetBlockListIQ;
+import org.jivesoftware.smackx.blocking.element.UnblockContactIQ;
+import org.jivesoftware.smackx.disco.ServiceDiscoveryManager;
+import org.jxmpp.jid.Jid;
+
+/**
+ * Blocking command manager class.
+ * 
+ * @author Fernando Ramirez
+ * @see <a href="http://xmpp.org/extensions/xep-0191.html">XEP-0191: Blocking
+ *      Command</a>
+ */
+public final class BlockingCommandManager extends Manager {
+
+    static {
+        XMPPConnectionRegistry.addConnectionCreationListener(new ConnectionCreationListener() {
+            @Override
+            public void connectionCreated(XMPPConnection connection) {
+                getInstanceFor(connection);
+            }
+        });
+    }
+
+    private static final Map<XMPPConnection, BlockingCommandManager> INSTANCES = new WeakHashMap<>();
+
+    /**
+     * Get the singleton instance of BlockingCommandManager.
+     *
+     * @param connection
+     * @return the instance of BlockingCommandManager
+     */
+    public static synchronized BlockingCommandManager getInstanceFor(XMPPConnection connection) {
+        BlockingCommandManager blockingCommandManager = INSTANCES.get(connection);
+
+        if (blockingCommandManager == null) {
+            blockingCommandManager = new BlockingCommandManager(connection);
+            INSTANCES.put(connection, blockingCommandManager);
+        }
+
+        return blockingCommandManager;
+    }
+
+    private BlockingCommandManager(XMPPConnection connection) {
```

Comment:
I don't think we need the full Manager pattern for BlockingCommand. It's such a simple XEP and the fact that the constructor does nothing further indicates that this is not needed. Simply make all methods static and add another parameter `XMPPConnection` to them. Look for at ClientStateIndicationManager, IIRC it does the same.